app/src/main/java/com/sanjeevyadavit/magecart/presentation/products/components/ProductList.kt

```diff
@@ -1,67 +1,61 @@
 package com.sanjeevyadavit.magecart.ui.products
 
 import android.util.Log
-import androidx.compose.foundation.Image
-import androidx.compose.foundation.background
 import androidx.compose.foundation.border
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.height
 import androidx.compose.foundation.layout.heightIn
 import androidx.compose.foundation.lazy.grid.GridCells
 import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
 import androidx.compose.foundation.lazy.grid.itemsIndexed
-import androidx.compose.foundation.lazy.grid.rememberLazyGridState
 import androidx.compose.material.CircularProgressIndicator
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.unit.dp
 import coil.compose.AsyncImage
-import com.sanjeevyadavit.magecart.model.Constants
-import com.sanjeevyadavit.magecart.model.StoreConfigs
-import com.sanjeevyadavit.magecart.model.product.ProductData
-import com.sanjeevyadavit.magecart.model.product.Products
-import com.sanjeevyadavit.magecart.service.RetrofitClient
-import com.sanjeevyadavit.magecart.viewmodel.ProductListViewModel
+import com.sanjeevyadavit.magecart.data.remote.dto.StoreConfigsDto
+import com.sanjeevyadavit.magecart.domain.model.Product
+import com.sanjeevyadavit.magecart.domain.model.StoreConfigs
 
 @Composable
-fun ProductList(viewModel: ProductListViewModel, storeConfigs: StoreConfigs? = null){
-    val loadMore = viewModel.loadMore.value
-    val products = viewModel.products.value
-
+fun ProductList(
+    products: List<Product>,
+    loadMore: Boolean,
+    totalCount: Int,
+    onScroll: (Int) -> Unit,
+    fetchMore: () -> Unit,
+    storeConfigs: StoreConfigs? = null
+) {
     Column {
         LazyVerticalGrid(columns = GridCells.Fixed(2)) {
             itemsIndexed(products) { index, item ->
                 // Question: This pagination code, does it make sense to be here?
-                viewModel.onChangeProductListScrollPosition(index)
-                if(index == products.size - 1 && (index + 1) < viewModel.getTotalCount() && !loadMore){
-                    viewModel.fetchNextPage()
+                onScroll(index)
+                if (index == products.size - 1 && (index + 1) < totalCount && !loadMore) {
+                    fetchMore()
```

Comment:
You can also use launched effect here. like the way we would have used useEffect here i guess

Reply:
Don't know what effect in compose is

Reply:
https://developer.android.com/jetpack/compose/side-effects